src/app_service/service/community/service.nim

```diff
@@ -1203,6 +1240,24 @@ QtObject:
 
     self.joinedCommunities[communityId] = community
 
+  proc cancelRequestToJoinCommunity*(self: Service, communityId: string) =
+    try:
+      var i = 0
+      for communityRequest in self.myCommunityRequests:
+        if (communityRequest.communityId == communityId):
+          let response = status_go.cancelRequestToJoinCommunity(communityRequest.id)
```

Comment:
Here should be emited AC notification: `self.activityCenterService.parseACNotificationResponse(response)`